### PR TITLE
Update baseline to 2.414.x

### PR DESCRIPTION
**Pull request by jglick**
The oldest suggested baseline according to https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/#currently-recommended-versions

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,5 +1,11 @@
-buildPlugin(useContainerAgent: true, configurations: [
-    [ platform: "linux", jdk: "11"],
-    [ platform: "linux", jdk: "17"],
-    [ platform: "windows", jdk: "11"]
+/*
```
Comment (jglick) [inline Jenkinsfile:1]: Updated from `archetypes`.

File: pom.xml [hunk 2]
```diff
@@ -76,8 +76,8 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.361.x</artifactId>
-        <version>2102.v854b_fec19c92</version>
+        <artifactId>bom-2.414.x</artifactId>
+        <version>2839.v003b_4d9d24fd</version>
```
Comment (jglick) [inline pom.xml:80]: Also currently latest BOM for this line.

**Comment by jglick** [comment]
Not sure what is causing the [`addContentsRunningBuild` flake](https://github.com/jenkinsci/support-core-plugin/pull/515/checks?check_run_id=22152102666). Looks like the sort of thing https://github.com/jenkinsci/workflow-support-plugin/pull/235 was supposed to have fixed. Not reproducible locally (including on Windows 10).

File: src/test/java/com/cloudbees/jenkins/support/impl/NodeExecutorsTest.java [hunk 3]
```diff
@@ -91,7 +96,7 @@ public void addContentsRunningBuild() throws Exception {
         WorkflowRun workflowRun = Optional.ofNullable(p.scheduleBuild2(0))
                 .orElseThrow(AssertionFailedError::new)
                 .waitForStart();
-        j.waitForMessage("Running on", workflowRun);
+        SemaphoreStep.waitForStart("wait/1", workflowRun);
```
Comment (jglick) [inline src/test/java/com/cloudbees/jenkins/support/impl/NodeExecutorsTest.java:99]: @cloudbees internal reference: BEE-41313

File: src/test/java/com/cloudbees/jenkins/support/impl/NodeExecutorsTest.java [hunk 4]
```diff
@@ -188,5 +193,8 @@ public void addContentsFiltered() throws Exception {
         assertTrue(executorMd.contains(filteredNodeDisplayName));
         assertTrue(executorMd.contains(filteredJobName));
         assertTrue(executorMd.contains(filteredJobDisplayName));
+
+        SemaphoreStep.success("wait/1", null);
+        j.waitForCompletion(workflowRun);
```
Comment (jglick) [inline src/test/java/com/cloudbees/jenkins/support/impl/NodeExecutorsTest.java:198]: BEE-46597

File: Jenkinsfile [hunk 5]
```diff
@@ -1,5 +1,11 @@
...
+ See the documentation for more options:
+ https://github.com/jenkins-infra/pipeline-library/
+*/
+buildPlugin(
+  forkCount: '1C', // run this number of tests in parallel for faster feedback.  If the number terminates with a 'C', the value will be multiplied by the number of available CPU cores
+  useContainerAgent: true, // Set to `false` if you need to use Docker for containerized tests
+  configurations: [
+    [platform: 'linux', jdk: 21],
+    [platform: 'windows', jdk: 17],
```
Comment (rsandell) [inline Jenkinsfile:10]: I am still baffled why this is the recommended setting. From a cost perspective sure. But I believe there should at least be one combination where just one variable is different.

**Review by rsandell — APPROVED** [review]

File: Jenkinsfile (hunk 5 above)
Comment (jglick) [reply Jenkinsfile:10]: This is not the place to discuss it; just copying the `Jenkinsfile` from `archetypes`.

**Pull request merged**